compute_tools/src/spec_apply.rs

```diff
@@ -120,36 +123,94 @@ where
 
         debug!("Applying phase {:?}", &apply_spec_phase);
 
-        let active_queries = ops
-            .map(|op| {
-                let Operation { comment, query } = op;
-                let inspan = match comment {
-                    None => span.clone(),
-                    Some(comment) => info_span!("phase {}: {}", comment),
-                };
-
-                async {
-                    let query = query;
-                    let res = client.simple_query(&query).await;
-                    debug!(
-                        "{} {}",
-                        if res.is_ok() {
-                            "successfully executed"
-                        } else {
-                            "failed to execute"
-                        },
-                        query
-                    );
-                    res
+        // TODO: Is there a way to write this with less nesting?
+        match apply_spec_phase {
+            ApplySpecPhase::RenameAndDeleteDatabases => {
+                // We need to run the operations sequentially, because we need to
+                // drop the database subscriptions before dropping the database.
+                for op in ops {
```

Comment:
looks like before this change there was a bug,
 because `unset_template_for_drop_dbs` was performed concurrently with `drop_db_query`

@MMeent , am I right, or there was some protection from the race condition, that I missed?

Reply:
Aha, it worked before because in your code there is only 1 await per operation and runtime doesn't have a point to switch to another task.
In my code, I added one more awaiting operation and that's why I got incorrect order of the operations in the test.